### PR TITLE
Added intersectSurface to NaifDskShape so the intercept point would b…

### PR DESCRIPTION
**Pull request by scsides**
Override NaifDskShape::intersectSurface so the intercept is calculated.

## Description

footprintinit segfaulted when setGround was called on the subSpacecraftPoint because the intercept destructor had been called after exiting a scope

## Related Issue
#4943

## Motivation and Context
Stop the segfault

## How Has This Been Tested?
footprintinit no longer segfaults and a test was added to make sure the intercept was being calculated

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [X] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by acpaquette — DISMISSED** [review]
Overall this looks good to me but having @KrisBecker and @jessemapel review would be ideal

File: isis/src/base/objs/NaifDskShape/unitTest.cpp [hunk 1]
```diff
@@ -80,6 +80,18 @@ int main(int argc, char *argv[]) {
     success = shapeModelFromPvlShape.intersectSurface(obsPos, lookDir);
     qDebug() << "Intersection successful?     " << toString(success);
 
+    qDebug() << "Try to intersect surface at obsPos (1000,0,0) and lookDir (-1,0,0)";
+    obsPos[0]  = 1000.0;  obsPos[1]  = 0.0;   obsPos[2]  = 0.0;
+    lookDir[0] = -1.0;  lookDir[1] = 0.0;  lookDir[2] = 0.0;
+    success = shapeModelFromPvlShape.intersectSurface(obsPos, lookDir);
```
Comment (jessemapel) [inline isis/src/base/objs/NaifDskShape/unitTest.cpp]: This is calling the wrong intersectSurface function, it should call the ground point + observer position version.

**Review by jessemapel — COMMENTED** [review]
I worked with Stuart on the change, so I don't have any input on the functional change. Just the test change.

**Review by KrisBecker — DISMISSED** [review]
This code is difficult to follow, top to bottom.

This looks like a good addition. Note that it costs two ray traces in this mod. One to determine the local radius at the lat/lon location and then a ray trace from observer to the surface point. That looks like a total of three ray traces/point so expect it to be a bit slow.

An alternative would be to call [ShapeModel::intersectSurface(lat,lon,obs,backcheck)](https://github.com/USGS-Astrogeology/ISIS3/blob/d8092c11645cab74c017e51c8c43ccb37db9bedc/isis/src/base/objs/ShapeModel/ShapeModel.cpp#L86-L108). This costs one ray trace but may result in a different intercept. Not sure it would be a different intercept in this context, but you would not need this update for this approach. Though, it would require a modification to [Camera::SetGround()](https://github.com/USGS-Astrogeology/ISIS3/blob/d8092c11645cab74c017e51c8c43ccb37db9bedc/isis/src/base/objs/CameraGroundMap/CameraGroundMap.cpp#L68-L100). 

Also note that the back projection check is ignored in this implementation. It may not be returning the same/valid intercept point as the surface point because of occlusion.

File: isis/src/base/objs/NaifDskShape/unitTest.cpp (hunk 1 above)
Comment (scsides) [reply isis/src/base/objs/NaifDskShape/unitTest.cpp]: Called other function

**Review by jessemapel — DISMISSED** [review]

**Review by jessemapel — APPROVED** [review]

**Pull request merged**